firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java

```diff
@@ -298,22 +298,16 @@ static class RateLimiterImpl {
      */
     synchronized boolean check(@NonNull PerfMetric metric) {
       Timer now = clock.getTime();
-      long newTokens =
-          Math.max(
-              0,
-              (long)
-                  (lastTimeTokenReplenished.getDurationMicros(now)
-                      * rate.getTokensPerSeconds()
-                      / MICROS_IN_A_SECOND));
-      tokenCount = Math.min(tokenCount + newTokens, capacity);
-      if (newTokens > 0) {
-        lastTimeTokenReplenished =
-            new Timer(
-                lastTimeTokenReplenished.getMicros()
-                    + (long) (newTokens * MICROS_IN_A_SECOND / rate.getTokensPerSeconds()));
+      double newTokens =
```

Comment:
A similar issue would exist in iOS as well. We might need to followup on that to fix it.

Reply:
Does iOS also have our own RateLimiter implementation?